### ISSUE TITLE
Requesting for IPv6 support for UE testcase list automated in 5G.

### ISSUE THREAD

**Issue by GANESH-WAVELABS** [magma/magma#9886]
Requesting for IPv6 support for UE testcase list automated in 5G.

**Comment by arun-magma** [comment]
Hi Ganesh:
Whats the Target date for this code readiness ? , Accordingly we can prioritize the scripts & test infra ?
Can you add the proposal link for this feature ?
Hope the need for IPv6 is only for UE IPv6 traffic only ?  notsure if GTP tunnel IP header and other N2 & N3 control plane have plans for IPv6 support , Kindly Clarify

**Comment by GANESH-WAVELABS** [comment]
Hi Arun,

Target end date for this feature is 30-Nov-2021 including testing. So it would be great if we have scripts ready by 17/18th of Nov-2021.
Proposal doc is still under preparation, will share once it is ready.
Yes this IPv6  is feature is only for UE, We got message like gNB IPv6 support is less priority for the time being, it is not part of this feature.

**Comment by arun-magma** [comment]
Whether N6 Interface supports IPv6 / Dual stack ?
Are we planning Dual stack UE Ipv4v6 support ?

**Comment by NeelimaKalpana** [comment]
Hi Ankur & Arun,

Please find some scenarios that we need  for IPV6 support.

Scenario 1: UE Requesting PDU-Type as IPV6
Step1:trigger PDU session with UE requesting session type=ipv6
Step2:PDU-Session should come up with IPV6 address proposed in the UE.
Step3. Try to send the IPV6 traffic and traffic should flow as expected.

Scenario 2: UE Requesting PDU-Type as IPV4V6
Step2:trigger PDU session with UE requesting session type=ipv4v6
Step3:PDU-Session should come up with IPV4V6 address proposed in the UE.
Step4. Try to send both IPV4 and IPV6 traffic and traffic should flow as expected.

Scenario 3: UE Requesting PDU-Type as IPV4 with one session and IPv6 with 2nd session
Step1:trigger PDU session with UE requesting session type=ipv4 and type=1pv6
Step2:PDU-Session should come up with IPV4 and IPv6  address proposed in respective sessions .
Step3. Try to send the IPV4 traffic and IPv6 traffic.

**Comment by arun-magma** [comment]
@GANESH-WAVELABS : Please confirm on N6 interface IPv6 address needs, if it need to be dual stack or not ?, this is critical info needed for setup prep.

**Comment by GANESH-WAVELABS** [comment]
@arun-magma  yes N6 interface also should support IPv6 address for UE, also dual stack IPV4V6 supporting as part of this feature. Please find the design doc which is under review:
https://docs.google.com/document/d/1yE862ImntChYpkLSDoSwXVB0yvtYCfl3wbP42VRH3w8/edit#

**Comment by arun-magma** [comment]
TC-62 Scripts available for IPv6 control plane verification, WIP for user plane dual stack update in eth0 (n6) for user traffic IPv6 verifications and arrange UE IPv6 IP pool with range < /64

**Comment by arun-magma** [comment]
May I know ETA for code readiness ?

**Comment by NeelimaKalpana** [comment]
Hi Arun,

"Automatically added this to Sprint: Nov 15 - Nov 26 based on Sprint automation 5 days ago"

Ideally the above sprint date will be the ETA for this feature

**Comment by arun-magma** [comment]
@NeelimaKalpana : Added based on your Request, lets validate once feature land in master.
TC63a	Verify 5G UE IPv4 only and Ipv6 only in two diff pdu sessions
